src/components/Form/FormDatePicker.react.js

```diff
@@ -0,0 +1,176 @@
+// @flow
+
+import * as React from "react";
+
+import FormSelect from "./FormSelect.react";
+import FormInputGroup from "./FormInputGroup.react";
+
+type Props = {|
+  children?: React.Node,
+  className?: string,
+  defaultDate: Date,
+  minYear: number,
+  maxYear: number,
+  format: string,
+  monthLabels: Array<string>,
+  onChange?: Function,
+|};
+
+type State = {|
+  currentDate: Date,
+|};
+
+type ChangeTypes = "mm" | "yyyy" | "dd";
+
+class FormDatePicker extends React.PureComponent<Props, State> {
+  static defaultProps = {
+    monthLabels: [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ],
+    minYear: 1897,
+    maxYear: new Date().getFullYear(),
+    format: "mm/dd/yyyy",
+    defaultDate: new Date(),
+  };
+
+  state = {
+    currentDate: new Date(),
+  };
+
+  // Set the default date from props
+  componentWillMount(): void {
+    const { defaultDate } = this.props;
+    this.setState({ currentDate: defaultDate });
+  }
+
+  // Handle date changes
+  _handleOnChange = (type: ChangeTypes, value: number): void => {
+    const { currentDate } = this.state;
+    const { onChange } = this.props;
+    const newDate: Date = new Date(currentDate);
+
+    // Change month
+    if (type === "mm") {
+      newDate.setMonth(value);
+    }
+
+    // Change day
+    if (type === "dd") {
+      newDate.setDate(value);
+    }
+
+    if (type === "yyyy") {
+      newDate.setFullYear(value);
+    }
+
+    this.setState({ currentDate: newDate }, () => {
+      onChange && onChange(this.state.currentDate);
+    });
+  };
+
+  // Creates an array with numeric values from start to end
+  _range = (start: number, end: number): Array<number> =>
+    Array.from({ length: end + 1 - start }, (v, k) => k + start);
+
+  // Renders the months select
+  _renderMonths = (): React.Node => {
+    const { currentDate } = this.state;
+    const { monthLabels } = this.props;
+
+    const onChangeMonths = ({ target }) =>
+      this._handleOnChange("mm", target.value);
+
+    return (
+      <FormSelect onChange={onChangeMonths}>
+        {monthLabels &&
```

Comment:
`monthLabels` is a required prop, thus it should never have a null / falsy value.  We can simplify `{monthLabels && monthLabels.map(...` to `{monthLabels.map(...`